File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/enums/DataEncryption.java
package com.shmedo.configlibrary.ble.enums;

/**
 * Created by adu on 2018/3/22.
 * 数据加密
 */
public enum DataEncryption {

    CLEAR(1),CIPHER(2);

    private int model;
    DataEncryption(int i) {
        this.model = i;
    }
    public int toInt() {
        return model;
    }

    public static DataEncryption valueOf(int model) {
        switch (model) {
            case 1: return CLEAR;
            case 2: return CIPHER;
            default: return CLEAR;
        }
    }
}

File: lib_cmd/src/main/java/com/shmedo/lib/cmd/base/md_cmd/enums/DataEncryption.kt
package com.shmedo.lib.cmd.base.md_cmd.enums

/**
 * Created by adu on 2018/3/22.
 * 数据加密
 */
enum class DataEncryption(private val model: Int) {
    CLEAR(1),
    CIPHER(2);

    fun toInt(): Int {
        return model
    }

    companion object {
        fun value(model: Int): DataEncryption {
            return when (model) {
                1 -> CLEAR
                2 -> CIPHER
                else -> CLEAR
            }
        }
    }
}
